### ISSUE TITLE
cell-var-from-loop ignores generator expressions

### ISSUE THREAD

**Issue by shestakovich** [pylint-dev/pylint#8221]
### Bug description

```python
"""cell-var-from-loop bug example."""
gens = []
loop_vars = [1, 2]
for loop_var in loop_vars:
    g = (value for value in [1, 2] if value == loop_var)
    gens.append(g)

for gen in gens:
    print(list(gen))
```
in the line `g = (value for value in [1, 2] if value == loop_var)` I'm expecting "cell-var-from-loop" warning, but pylint ignores it

### Command used

```shell
pylint main.py
```

### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior
```shell
************* Module main
main.py:5:48: W0640: Cell variable loop_var defined in loop (cell-var-from-loop)

------------------------------------------------------------------
Your code has been rated at 8.89/10 (previous run: 8.89/10, +0.00)
```

### Pylint version

```shell
pylint 2.16.1
astroid 2.14.1
Python 3.11.1 (main, Jan  6 2023, 00:00:00) [GCC 12.2.1 20221121 (Red Hat 12.2.1-4)]
```

### OS / Environment

Fedora 37
